src/sentry/api/endpoints/organization_repositories.py

```diff
@@ -54,21 +56,24 @@ def get(self, request, organization):
         # TODO(mn): Remove once old Plugins are removed or everyone migrates to
         # the new Integrations. Hopefully someday?
         elif status == 'unmigratable':
-            repos = []
-
             integrations = Integration.objects.filter(
                 organizationintegration__organization=organization,
                 organizationintegration__status=ObjectStatus.ACTIVE,
                 provider__in=('bitbucket', 'github', 'vsts'),
                 status=ObjectStatus.ACTIVE,
             )
 
-            repos = [
-                repo
-                for i in integrations
-                for repo in i.get_installation(organization.id)
-                             .get_unmigratable_repositories()
-            ]
+            repos = []
+
+            for i in integrations:
+                try:
+                    repos.extend(i.get_installation(organization.id)
+                                  .get_unmigratable_repositories())
+                except ApiError:
+                    raven.captureException()
```

Comment:
If we log this to Sentry, what are we going to do with this information? It doesn't seem like it'd be something we can fix. Seems like it'd be a misconfiguration on their end. I'd more argue that we return this information in the response body rather than reporting the error to Sentry.

Reply:
@mattrobenolt 

> I guess you're also doing this in a loop for each integration. So that'd be really bad. Can we avoid making HTTP requests in a loop in the API endpoint?

We're doing it once per Integration, where each one will be a different external service, so we can't really prevent the looping aspect.

> If we log this to Sentry, what are we going to do with this information? It doesn't seem like it'd be something we can fix. Seems like it'd be a misconfiguration on their end. I'd more argue that we return this information in the response body rather than reporting the error to Sentry.

It just gives us insight into what's going on when things go wrong. Can tell us whether it's a timeout, bad request, whatever.

The point of reporting it and moving on is to not make the entire page fail when one service has issues. It's fine to gracefully fail here.

Reply:
Context: this is happening after an Integration is installed. We're checking whether we can access all of the Repositories from our DB, on GitHub/VSTS/etc., using the new Integration's creds.

There isn't really anything we can check on our side, since it's all about what we have access to on the other side. We could maybe store a list of accessible repos, but that list is mutable and seems kind of silly to replicate.

> And for reporting into Sentry, the only argument I have against it is... it's going to alert people (like me), and this isn't something actionable to fix. I think you just want logging here instead so we can dig it out of there.

As long as we have some info about what happened, somewhere accessible to people, I'm happy. Debugging distributed shit is super annoying/impossible when running blind. Even if it's not actionable, it's useful to know that. Can we tag things in some way that notifies the right people instead of you?

Reply:
You can add ownership rules. :)